src/config.rs

```diff
@@ -441,11 +441,18 @@ impl Config {
         })
     }
 
-    pub fn load(enable_styles: bool) -> Result<Self> {
+    /// Load and read the config file into a [Config] and return it.
+    ///
+    /// path: The path to the config file could be specified with the path variable.
+    /// if not specified, the default config file will be determined and used.
+    ///
+    /// enable_styles: Whether to use colors, etc.
+    pub fn load(path: Option<&Path>, enable_styles: bool) -> Result<Self> {
```

Comment:
So I think we should just have `Config::load(path: &Path, enable_styles: bool) -> Result<Self>`. Then, we change `main` to do something like
```rust
let config = match args.config {
    Some(path) => Config::load(path, enable_styles)?,
    None => Config::load_default_location(enable_styles)?,
};
```
where `load_default_location` checks whether there exists a file at the default location (like right now), but `load` errors if the file does not exist. You might want to introduce another private method to avoid some duplication here